Software/scan_key/AuthenticationSQL.py

```diff
@@ -1,42 +1,42 @@
 from on_off import *
 from ErrorSQL import *
 import os
+import subprocess
 
 def machineAuth(id, cursor):
-    os.system("hostname > tmp")
-    currMachineID = open('tmp', 'r').read()
-    currMachineID = currMachineID.split()
+    currMachineID = subprocess.check_output(['hostname'])
+    name = name.strip()
 
-    cursor.execute("select machineID, machineType from MACHINE")
+    cursor.execute("select machineID, machineType FROM MACHINE WHERE t1String = " + id)
```

Comment:
The table MACHINE does not contain t1String.

The idea here is you need to fetch the machineType for the machineID we just got from subprocess.